### DISCUSSION TITLE
issue: defaultValues not working with FormProvider and useFormContext

### DISCUSSION THREAD

**Discussion by efraimrodrigues** [react-hook-form/react-hook-form#11478] [Q&A]
### Version Number

7.0.0-alpha.0 & 7.50.0

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-form-normalize-field-forked-khppy8?file=%2Fsrc%2Findex.js%3A69%2C41

### Steps to reproduce

1. Load the CodeSandbox Link
2. Note `Material UI Price` doesn't load with defaultValue

### Expected behaviour

`Material UI Price` input should load with defaultValue

I have updated [Normalize/Format/Mask Field](https://github.com/react-hook-form/react-hook-form/blob/master/examples/V7/normalizeField.tsx) example a little bit to mimic the configuration I have in my project and use `FormProvider` and `useFormContext` to create a controlled input. I expected the `Material UI Price` input to load with default value but it doesn't. 

Although I can see the `ref` object getting into `TextField`, I have the impression this has to do with `ref` not being managed or used correctly. I have tried different ways of giving it to the `customInput` (see example in the CodeSandbox link) but none of them worked. 

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
No log output. No errors in the console.
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by chanthinh** [comment]
Hi @efraimrodrigues, 
This is my codesandbox: https://codesandbox.io/p/sandbox/react-hook-form-normalize-field-forked-3x7qkx?file=%2Fsrc%2Findex.js%3A75%2C31
You can check it

**Reply by efraimrodrigues** [reply]
Hi @chanthinh . 

Thank you for the workaround. It does work with that. I wonder if we should expect it to work with register as well.

**Reply by efraimrodrigues** [reply]
@bluebill1049 can I try fixing this (i.e. making it work with `register`) and submitting a PR?
